src/DynamoAnalyzer/Helper/GregHelper.cs

```diff
@@ -0,0 +1,241 @@
+using DynamoAnalyzer.Models;
+using Greg;
+using Greg.Requests;
+using Greg.Responses;
+
+namespace DynamoAnalyzer.Helper
+{
+    /// <summary>
+    /// Provides methos to manage packages
+    /// </summary>
+    public static class GregHelper
+    {
+        private static GregClient _client;
+
+        public static GregClient GetClient()
+        {
+            return _client ??= new GregClient(new Dynamo.Core.IDSDKManager(), ConfigHelper.GetConfiguration()["DynamoPackagesURL"]);
+        }
+
+        /// <summary>
+        /// Downloads the full package list
+        /// </summary>
+        /// <returns></returns>
+        public static async Task<List<PackageHeader>> GetPackages()
+        {
+            ResponseWithContentBody<List<PackageHeader>> pkgResponse = null;
+
+            LogHelper.Log("PackageList", "Download start");
+            await Task.Run(() =>
+            {
+                pkgResponse = GetClient().ExecuteAndDeserializeWithContent<List<PackageHeader>>(HeaderCollectionDownload.ByEngine(ConfigHelper.GetConfiguration()["Engine"]));
+            });
+
+            CleanPackagesWithWrongVersions(pkgResponse.content);
+
+            List<PackageHeader> response = pkgResponse.content.Where(f => !f.banned).ToList();
+
+            LogHelper.Log("PackageList", "Download end");
+
+            return response;
+        }
+
+        /// <summary>
+        /// For every package Checks its versions and exclude the ones with errors
+        /// </summary>
+        /// <param name="packages"></param>
+        static void CleanPackagesWithWrongVersions(List<PackageHeader> packages)
+        {
+            foreach (var package in packages)
+            {
+                bool packageHasWrongVersion = package.versions.Count(v => v.url == null) > 0;
+
+                if (packageHasWrongVersion)
```

Comment:
what does a wrong version mean?

Reply:
A package could have n number of versions, a wrong package version occurs when there is no url to download an specific version, this function can be found at Dynamo\src\DynamoPackages\PackageManagerClient.cs:L97